### DISCUSSION TITLE
How to set new points for Polygon ?

### DISCUSSION THREAD

**Discussion by senschen** [fabricjs/fabric.js#9115] [Q&A] [answered]
I did it by this way, but the shape will be clip
![image](https://github.com/fabricjs/fabric.js/assets/8544826/562d1c70-98d5-4753-a4fb-8c22a7e31b35)

```
target.set({
   points: path as fabric.Point[],
});

target.setCoords();
```

if I try to modify width and height to fix the clip, the left and top seems need to be modify together,
so I did this：
```
target.set({
   points: path as fabric.Point[],
   width: newWidth,
   height: newHeigt,
       
   left: left - newWidth, / 2,
   top: top - newWidth, / 2,
});

target.setCoords();
```
but the render is not right, too.
![image](https://github.com/fabricjs/fabric.js/assets/8544826/4cd13971-6ded-452d-9704-544d164b37d5)

**Comment by ShaMan123** [comment, marked as answer]
setDimensions if I am not mistaken
